Packet++/src/BgpLayer.cpp

```diff
@@ -703,6 +703,26 @@ void BgpUpdateMessageLayer::getNetworkLayerReachabilityInfo(std::vector<prefix_a
 	parsePrefixAndIPData(dataPtr, nlriSize, nlri);
 }
 
+bool BgpUpdateMessageLayer::isDataValid(const uint8_t *data, size_t dataSize)
+{
+	if (data == nullptr || dataSize < sizeof(bgp_common_header) + 2*sizeof(uint16_t))
+		return false;
+
+	uint16_t messageLen = be16toh(((bgp_common_header*)data)->length);
+	if (dataSize < messageLen)
+		return false;
```

Comment:
Some of the checks here can move to `BgpLayer::parseBgpLayer`, for example:
- Check if `data == nullptr`
- Check if `dataSize < messageLen`